plugin-build/src/main/kotlin/io/sentry/SentryKotlinCompilerGradlePlugin.kt

```diff
@@ -0,0 +1,34 @@
+package io.sentry
+
+import org.gradle.api.provider.Provider
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilation
+import org.jetbrains.kotlin.gradle.plugin.KotlinCompilerPluginSupportPlugin
+import org.jetbrains.kotlin.gradle.plugin.SubpluginArtifact
+import org.jetbrains.kotlin.gradle.plugin.SubpluginOption
+
+class SentryKotlinCompilerGradlePlugin : KotlinCompilerPluginSupportPlugin {
+
+    override fun isApplicable(kotlinCompilation: KotlinCompilation<*>): Boolean = true
+
+    override fun getCompilerPluginId(): String = getPluginArtifact().let {
+        "${it.groupId}.${it.artifactId}"
+    }
+
+    override fun getPluginArtifact(): SubpluginArtifact {
+        // needs to match sentry-kotlin-compiler-plugin/build.gradle.kts
+        return SubpluginArtifact(
+            groupId = "io.sentry",
+            artifactId = "sentry-kotlin-compiler-plugin",
+            version = "1.0.0-SNAPSHOT"
```

Comment:
as discussed internally, let's align with the android gradle plugin version. Ideally we'd have a single source of truth for the version for all artifacts (android-gradle-plugin,kotlin-compiler-gradle,kotlin-compiler). Maybe we could use the [buildkonfig](https://github.com/gmazzo/gradle-buildconfig-plugin) plugin to inject the version from gradle.properties if there's no other way